compiler/rustc_session/src/options.rs

```diff
@@ -825,6 +828,8 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
     dep_tasks: bool = (false, parse_bool, [UNTRACKED],
         "print tasks that execute and the color their dep node gets (requires debug build) \
         (default: no)"),
+    trim_diagnostic_paths: bool = (true, parse_bool, [UNTRACKED],
+        "in diagnostics, enable shortening of paths to unique symbols"),
```

Comment:
```suggestion
        "in diagnostics, use heuristics to shorten paths referring to items"),
```
Or something like that.
"Unique symbols" are an implementation detail.